pgrx/src/spi/tuple.rs

```diff
@@ -76,9 +113,12 @@ impl<'client> SpiTupleTable<'client> {
     fn get_spi_tuptable(
         &self,
     ) -> SpiResult<(*mut pg_sys::SPITupleTable, *mut pg_sys::TupleDescData)> {
-        let table = self.table.as_deref().ok_or(SpiError::NoTupleTable)?;
-        // SAFETY:  we just assured that `table` is not null
-        Ok((table as *const _ as *mut _, table.tupdesc))
+        let table = self.table.map(|table| table.as_ptr()).ok_or(SpiError::NoTupleTable)?;
+        let tupdesc = unsafe {
+            // SAFETY:  we just assured that `table` is not null
+            table.as_mut().unwrap().tupdesc
+        };
+        Ok((table, tupdesc))
```

Comment:
Is it actually possible for this pointer to be NULL?

Reply:
yes.  It is possible for `pg_sys::SPI_tuptable` to be NULL at the end of a successful statement.  There's some comments around that here:  https://github.com/eeeebbbbrrrr/pgrx/blob/b474516d777f8796f10caddfee718e1468f722b0/pgrx/src/spi/tuple.rs#L126-L130

Maybe they're in the wrong place.

Reply:
oh, but I suppose... to actually answer your question, in the cases where this function is called, no, it shouldn't be possible for for `self.table` to be NULL.  Sorry.

But I don't know that we can make that true such that we can elide the Option around the NotNull

Reply:
It would be an exceptional condition for it to be NULL -- made it should just panic in that case?

Not sure if you're asking because you see/know something I don't or just forcing me to think about it even more.

Reply:
Just trying to figure out for myself if the Option is actually mandatory, here.